OpenRA.Mods.Common/Widgets/Logic/Ingame/Hotkeys/CycleHarvestersHotkeyLogic.cs

```diff
@@ -0,0 +1,63 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Lint;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic.Ingame
+{
+	[ChromeLogicArgsHotkeys("CycleHarvestersKey")]
+	public class CycleHarvestersHotkeyLogic : SingleHotkeyBaseLogic
+	{
+		readonly Viewport viewport;
+		readonly ISelection selection;
+		readonly World world;
+
+		[ObjectCreator.UseCtor]
+		public CycleHarvestersHotkeyLogic(Widget widget, ModData modData, WorldRenderer worldRenderer, World world, Dictionary<string, MiniYaml> logicArgs)
+			: base(widget, modData, "CycleHarvestersKey", "WORLD_KEYHANDLER", logicArgs)
+		{
+			viewport = worldRenderer.Viewport;
+			selection = world.Selection;
+			this.world = world;
+		}
+
+		protected override bool OnHotkeyActivated(KeyInput e)
+		{
+			var player = world.RenderPlayer ?? world.LocalPlayer;
+
+			var harvesters = world.ActorsHavingTrait<Harvester>()
+				.Where(a => a.Owner == player)
```

Comment:
Adding a check for `a.IsInWorld` here will probably fix the issue with carryalls.